gcapi/gcapi.py

```diff
@@ -21,41 +26,80 @@ def is_uuid(s):
 
 
 class APIBase:
-    _client = None # type: Client
+    _client = None  # type: Client
     base_path = ""
     sub_apis = {}
 
+    json_schema_file = None
+    __json_schema = None
+
+    @staticmethod
+    def _attempt_load_json_schema(filename):
+        if filename is None:
+            return None
+        filename = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)), "schemas", filename
+        )
+
+        try:
+            with open(filename, "r") as f:
+                jsn = json.load(f)
+            return jsonschema.Draft7Validator(jsn)
+        except ValueError as e:
+            print("Error parsing '{}': {}".format(e, filename), file=sys.stderr)
+            return None
```

Comment:
Should this function return `None` here? Wouldn't it be better to raise an error here and let the callers handle it?